climpred_skeleton/comparison.py

```diff
@@ -0,0 +1,95 @@
+from typing import Union
+
+import pkg_resources
+import xarray as xr
+import yaml
+
+from .core import Verification
+
+
+class Comparison(Verification):
+    """First step in pipeline. Broadcast your prediction and observations
+    in a certain manner.
+
+    Base Comparison class hints at which attributes are expected for each
+    sub comparison class. It also stores the factory for internal use, to
+    return the appropriate subfunction.
+    """
+
+    _comparison_method: str
+    _hindcast_comparison: bool
+    _probabilistic_comparison: bool
+
+    def __init__(self, prediction: xr.Dataset, observation: xr.Dataset):
+        super().__init__(prediction, observation)
+
+    def get_comparison(self, method) -> 'Comparison':
+        try:
+            return comparison_dict[method](self._prediction, self._observation)
```

Comment:
is this the same as what we do now in climpred?

Reply:
This is similar to `utils.get_comparison_class` but a little cleaner. Here the base class is returning the subclass we need. If you view the pipeline picture I have in the original PR post, `Comparison` funnels down into `LeadAlignment` and will funnel into `Scoring`. Essentially this will be handled internally but at the `LeadAlignment` or `Scoring` step (for PM) you'll use the base `Comparison` class with a string (`'e2o'` or something) to return the subclass you need then run `.broadcast()` to get the ensembles set up like you need. Then you don't have to call `get_comparison_class()` in every function but instead this is done at the beginning of the pipeline every time.